client/components/edit/children/ChildList.tsx

```diff
@@ -1,52 +1,43 @@
 import React, { useEffect, useState } from "react";
-import { useFetchContext } from "../../../context/FetchContext";
-import { getObjectChildrenUrl } from "../../../util/routes";
+import { useChildListContext } from "../../../context/ChildListContext";
 import Child from "./Child";
 import Pagination from "@mui/material/Pagination";
 
 export interface ChildListProps {
-    pid: string;
-    pageSize: number;
-}
-interface Children {
-    numFound: number;
-    start: number;
-    docs: Record<string, string>[];
+    pid?: string;
+    pageSize?: number;
 }
 
 export const ChildList = ({ pid = "", pageSize = 10 }: ChildListProps): React.ReactElement => {
     const {
-        action: { fetchJSON },
-    } = useFetchContext();
-    const [children, setChildren] = useState<Children>({
-        numFound: 0,
-        start: 0,
-        docs: [],
-    });
+        state: { childStorage },
+        action: { getChildStorageKey, loadChildrenIntoStorage },
+    } = useChildListContext();
     const [page, setPage] = useState<number>(1);
-
+    const key = getChildStorageKey(pid, page, pageSize);
+    const notLoaded = typeof childStorage[key] === "undefined";
```

Comment:
Could you instead do childStorage.hasOwnProperty(key)?

Reply:
Good point. It also is more readable if the variable is called `loaded` instead of `notLoaded`. Revised here: [1d26e5f](https://github.com/FalveyLibraryTechnology/VuDLPrepJS/pull/140/commits/1d26e5fca56b1373ff213a0765f5c4b8c4e6eab6)